src/tox_lsr/test_scripts/runansible-test.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# SPDX-License-Identifier: MIT
+
+# A shell wrapper around ansible-test.
+
+set -u
+
+ME=$(basename "$0")
+SCRIPTDIR=$(readlink -f "$(dirname "$0")")
+
+. "$SCRIPTDIR/utils.sh"
+
+if [ "${LSR_ROLE2COLL_NAMESPACE:-}" = "" -o "${LSR_ROLE2COLL_NAME:-}" = "" ]; then
+    lsr_info "${ME}: Collection format only." 
+	exit 1
+fi
+
+LSR_ROLE2COLL_TOP="$MY_LSR_TOX_ENV_DIR"/ansible_collections/"$LSR_ROLE2COLL_NAMESPACE"/"$LSR_ROLE2COLL_NAME"
+
+rval=0
+if [ -d "$LSR_ROLE2COLL_TOP" ]; then
+  cd "$LSR_ROLE2COLL_TOP"
+  for test in $( ansible-test sanity --list-tests ); do
+    lsr_info "${ME}: Running $test ..."
+    ansible-test sanity --test $test; r=$? || :
```

Comment:
The `|| :` here is in the wrong place - I think you want something like this:
```bash
     r=0
     ansible-test sanity --test $test || r=$?
     rval=$( expr $rval + $r ) || :
```
This will set `r=$?` if `ansible-test` exits with a non-zero exit code, and `r` will get the value of that exit code.  Note that if `ansible-test` returns zero, the code after the `||` will not be executed, so we have to reset `r=0` every time through the loop.  Also note that `expr 0 + 0` will return a non-zero exit code to the shell, so we have to use `|| :` to ignore that.